tests/puppeteer/pages/BO/customers/add.js

```diff
@@ -47,7 +48,15 @@ module.exports = class AddCustomer extends BOBasePage {
     else await this.page.click(this.enabledSwitchlabel.replace('%ID', '0'));
     if (customerData.partnerOffers) await this.page.click(this.partnerOffersSwitchlabel.replace('%ID', '1'));
     else await this.page.click(this.partnerOffersSwitchlabel.replace('%ID', '0'));
-    await this.page.click(this.selectAllGroupAccessCheckbox);
+    if (customerData.defaultCustomerGroup === 'Customer') {
+      await this.page.click(this.selectAllGroupAccessCheckbox);
+    } else if (customerData.defaultCustomerGroup === 'Visitor') {
+      await this.page.click(this.groupAccessCheckkbox.replace('%ID', 1));
+      await this.page.click(this.groupAccessCheckkbox.replace('%ID', 2));
+    } else {
+      await this.page.click(this.groupAccessCheckkbox.replace('%ID', 0));
+      await this.page.click(this.groupAccessCheckkbox.replace('%ID', 2));
+    }
```

Comment:
```suggestion
    switch (customerData.defaultCustomerGroup) {
      case 'Customer':
        await this.page.click(this.selectAllGroupAccessCheckbox);
        break;
      case 'Visitor':
        await this.page.click(this.groupAccessCheckkbox.replace('%ID', 1));
        await this.page.click(this.groupAccessCheckkbox.replace('%ID', 2));
        break;
      case 'Guest':
        await this.page.click(this.groupAccessCheckkbox.replace('%ID', 0));
        await this.page.click(this.groupAccessCheckkbox.replace('%ID', 2));
        break;
      default :
        throw new Error(`${customerData.defaultCustomerGroup} was not found as a group access`);
    }
```
You should throw an error if group access was not found, WDYT ?

Reply:
I think link this will not work for edit customer and maybe the solution is to check if checkbox is selected before click on it.

Reply:
Done thx :)